plone/namedfile/file.py

```diff
@@ -266,14 +273,24 @@ class NamedImage(NamedFile):
     filename = FieldProperty(INamedFile['filename'])
 
     def __init__(self, data='', contentType='', filename=None):
-        self.contentType, self._width, self._height = getImageInfo(data)
-        self.data = data
         self.filename = filename
+        self._setData(data)
 
         # Allow override of the image sniffer
         if contentType:
```

Comment:
Setting of `self.contentType` is still missing, so that tests like https://github.com/plone/plone.namedfile/blob/master/plone/namedfile/tests/test_image.py#L41 are failing.
Either `contentType` needs to be set here to the passed in value `contentType`:

    if contentType or getattr(self, 'content_type', None):
        self.contentType = contentType

or, this needs to happen in ` _setData`, that means instead of

    def _setData(self, data):
        super(NamedImage, self)._setData(data)

        contentType, self._width, self._height = getImageInfo(self._data)
        if contentType:
            self.contentType = contentType

it would become

    def _setData(self, data):
        super(NamedImage, self)._setData(data)

        self.contentType, self._width, self._height = getImageInfo(self._data)